eZ/Bundle/EzPublishCoreBundle/Command/RemoveVersionsCommand.php

```diff
@@ -0,0 +1,232 @@
+            throw new InvalidArgumentException(
+                'status',
+                'Keep value can not be negative.'
+            );
+        }
+
+        $status = $input->getOption('status');
+
+        $contentIds = $this->getObjectsIds($keep, $status);
+        $contentIdsCount = count($contentIds);
+
+        if ($contentIdsCount === 0) {
+            $output->writeln('<info>There is no Contents matching given criteria.</info>');
+
+            return;
+        }
+
+        $output->writeln(sprintf(
+            '<info>Found %d Content IDs matching given criteria.</info>',
+            $contentIdsCount
+        ));
+
+        $removedVersionsCounter = 0;
+
+        foreach ($contentIds as $contentData) {
+            try {
+                $contentInfo = $this->contentService->loadContentInfo($contentData['id']);
+                $versions = $this->contentService->loadVersions($contentInfo);
+                $versionsCount = count($versions);
+
+                $output->writeln(sprintf(
+                    '<info>Content %d has %d version(s)</info>',
+                    $contentData['id'],
+                    $versionsCount
+                ), Output::VERBOSITY_VERBOSE);
+
+                $versions = array_slice(array_reverse($versions), $keep);
+
+                foreach ($versions as $version) {
+                    if (
+                        (!$version->isPublished() && $status === self::VERSION_ALL) ||
+                        ($version->isDraft() && $status === self::VERSION_DRAFT) ||
+                        ($version->isArchived() && $status === self::VERSION_ARCHIVED)
+                    ) {
+                        $this->contentService->deleteVersion($version);
+                        ++$removedVersionsCounter;
+                        $output->writeln(sprintf(
+                            "Content's (%s) version (%s) has been deleted.",
+                            $contentInfo->id,
+                            $version->id
+                        ), Output::VERBOSITY_VERBOSE);
+                    }
+                }
+            } catch (Exception $e) {
+                $output->writeln(sprintf(
+                    '<error>%s</error>',
+                    $e->getMessage()
+                ));
+            }
+        }
+
+        $output->writeln(sprintf(
+            '<info>Removed %d unwanted contents version(s).</info>',
+            $removedVersionsCounter
+        ));
+    }
+
+    /**
+     * @param int $keep
+     * @param string $status
+     *
+     * @return array
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     */
+    protected function getObjectsIds(int $keep, string $status): array
+    {
+        $query = $this->connection->createQueryBuilder()
+                ->select('c.id, v.status')
+                ->from('ezcontentobject', 'c')
+                ->leftJoin('c', 'ezcontentobject_version', 'v', 'v.contentobject_id = c.id')
```

Comment:
any reason for using `LEFT JOIN` instead of just `JOIN` (inner)? It would imply we expect to have `ezcontentobject` entries without any corresponding `ezcontentobject_version` entries.